src/Evolution/Systems/Cce/Events/ObserveFields.hpp

```diff
@@ -0,0 +1,421 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <pup.h>
+#include <string>
+#include <tuple>
+#include <type_traits>
+#include <unordered_set>
+#include <vector>
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/ComplexModalVector.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Evolution/Systems/Cce/IntegrandInputSteps.hpp"
+#include "Evolution/Systems/Cce/OptionTags.hpp"
+#include "Evolution/Systems/Cce/Tags.hpp"
+#include "IO/Observer/ObserverComponent.hpp"
+#include "IO/Observer/ReductionActions.hpp"
+#include "NumericalAlgorithms/Spectral/SwshCoefficients.hpp"
+#include "NumericalAlgorithms/Spectral/SwshCollocation.hpp"
+#include "NumericalAlgorithms/Spectral/SwshTransform.hpp"
+#include "Options/Context.hpp"
+#include "Options/ParseError.hpp"
+#include "Options/String.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Event.hpp"
+#include "Time/Tags/Time.hpp"
+#include "Utilities/MakeString.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits/IsA.hpp"
+
+/// \cond
+template <size_t Dim>
+class Mesh;
+namespace Frame {
+struct Inertial;
+}  // namespace Frame
+/// \endcond
+
+namespace Cce {
+namespace Events {
+namespace detail {
+template <typename Tag>
+std::string name() {
+  if constexpr (std::is_same_v<Tag, Tags::ComplexInertialRetardedTime>) {
+    return db::tag_name<Tags::InertialRetardedTime>();
+  } else {
+    return db::tag_name<Tag>();
+  }
+}
+}  // namespace detail
+
+/*!
+ * \brief Event to observe fields/variables in a characteristic evolution.
+ *
+ * \details Similar to `dg::Events::ObserveFields`, this event will write volume
+ * data from the characteristic domain to disk when triggered. However, there a
+ * several differences which are important to highlight.
+ *
+ * First is the fields themselves. The DG event takes the fields to observe as
+ * template parameters because the event must work with many evolution systems.
+ * However, since this event is specific to the characteristic evolution system,
+ * we can hardcode the list of fields that are available to observe. The fields
+ * available to observe are the following tags along with their first and second
+ * `Cce::Tags::Dy` derivatives (see `Cce::Tags::Dy` for a definition of y`):
```

Comment:
You have an unmatched backtick at the end.  Not sure if you are missing any other intended characters as well.